pkg/services/grpcserver/interceptors/auth.go

```diff
@@ -80,42 +86,37 @@ func (a *authenticator) tokenAuth(ctx context.Context) (context.Context, error)
 }
 
 func (a *authenticator) getSignedInUser(ctx context.Context, token string) (*user.SignedInUser, error) {
-	decoded, err := apikeygenprefix.Decode(token)
+	claims, err := a.PluginAuthService.Verify(ctx, token)
 	if err != nil {
 		return nil, err
 	}
 
-	hash, err := decoded.Hash()
-	if err != nil {
-		return nil, err
+	subject, ok := claims["sub"].(string)
+	if !ok || subject == "" {
+		return nil, status.Error(codes.Unauthenticated, "token missing subject claim")
 	}
 
-	apikey, err := a.APIKeyService.GetAPIKeyByHash(ctx, hash)
-	if err != nil {
-		return nil, err
+	userInfo := xctx.UserInfoFromString(subject)
+	if userInfo == nil {
+		return nil, status.Error(codes.Unauthenticated, "invalid subject claim")
 	}
 
-	if apikey == nil || apikey.ServiceAccountId == nil {
-		return nil, status.Error(codes.Unauthenticated, "api key does not have a service account")
+	if userInfo.UserID == 0 {
```

Comment:
i am not sure if `userId === 0` always means anonymous access, e.g. alerting takes screenshots with `userId: 0` https://github.com/grafana/grafana/blob/bad4f7fec5f479adce1ce0055148b33370a8f072/pkg/services/screenshot/screenshot.go#L104-L106 - not sure if there are other similar use cases

Reply:
good catch. i attempted to solve this a bit differently in https://github.com/grafana/grafana/pull/57420/commits/c43a902af54252053c8ed752555ef8e102f3a63a